subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaPlugin.java

```diff
@@ -50,27 +51,34 @@ public void apply(Project project) {
         project.getPluginManager().apply(JavaPlugin.class);
 
         final SourceSet main = project.getExtensions().getByType(JavaPluginExtension.class).getSourceSets().getByName("main");
-
-        configureScaladoc(project, main);
-
         final Configuration incrementalAnalysisElements = project.getConfigurations().getByName("incrementalScalaAnalysisElements");
         String compileTaskName = main.getCompileTaskName("scala");
         final TaskProvider<AbstractScalaCompile> compileScala = project.getTasks().withType(AbstractScalaCompile.class).named(compileTaskName);
         final Provider<RegularFile> compileScalaMapping = project.getLayout().getBuildDirectory().file("tmp/scala/compilerAnalysis/" + compileTaskName + ".mapping");
         compileScala.configure(task -> task.getAnalysisMappingFile().set(compileScalaMapping));
         incrementalAnalysisElements.getOutgoing().artifact(
             compileScalaMapping, configurablePublishArtifact -> configurablePublishArtifact.builtBy(compileScala));
+            
+        configureScaladoc(project, main, compileScala);
     }
 
-    private static void configureScaladoc(final Project project, final SourceSet main) {
+    private static void configureScaladoc(final Project project, final SourceSet main, final TaskProvider<AbstractScalaCompile> compileScala) {
```

Comment:
`compileScala` doesn't seem to be used, should we make the scaladoc task depend on it if we use Scala 3 ?

Reply:
Leftover, will remove